server/api/documents.js

```diff
@@ -7,16 +7,37 @@ router.get(
   "/",
   passport.authenticate("jwt", { session: false }),
   (req, res) => {
-    docsDb
-      .getAllDocuments()
-      .then(data => {
-        res.send(data);
-      })
-      .catch(err => {
-        console.error(err);
-        res.send(500);
-      });
+    let user = req.user.role;
+    console.log(user);
+
+    if (user) {
+      getIterpreterDocuments(res);
+    } else {
+      getUserDocuments();
+    }
   }
 );
 
+function getIterpreterDocuments(res) {
+  docsDb
+    .getAllDocuments()
+    .then(data => {
+      res.send(data);
+    })
+    .catch(err => {
+      console.error(err);
+      res.send(500);
+    });
+}
+function getUserDocuments(res) {
```

Comment:
These 2 functions getInterpreterDocuments and getUserDocuments are doing exactly the same things, just call different functions from docsDb, can you simplify it to avoid code duplication?

Reply:
How about something like this:

```
let getDocumentFunction;
if(role === INTERPRETER) {
  getDocumentFunction = docsDb.getAllDocuments;
} else {
  getDocumentFunction = docsDb.getUserDocuments;
}

getDocumentFunction().then(data => {
 res.send(data);
}).catch(err => {
  res.send(500);
};
```